### ISSUE TITLE
Consider adopting a Brouwerian interpretation of "mental construction"

### ISSUE THREAD

**Issue by jonsterling** [jonsterling/agda-bar-induction#11]
I think it may be possible to prove the bar theorem directly if we switch from Martin-Löf's interpretation of the quantifiers to Brouwer's—I believe, in the case of quantifying over choice sequences as in `forall α. P(α)`, such a mental construction is no longer merely an arbitrary computational function from streams to the proofs that they satisfy `P`, but is instead a _neighborhood function_, which is a function from finite approximations of `α` to either "I need more info" or a construction that represents the immediate intuition that `P` holds for every choice sequence in the neighborhood under consideration.

As a result, the demonstration that a node is barred will already be well-ordered, and so the bar theorem itself ought to be an "obvious truth".

**Comment by ghost** [comment]
Hmm, this sounds a little bit like how the quantifiers might work if interpreted in terms of Kan extensions. Ran is a bit like a generalized continuation type and this is how I was using it (with the stage monad) back when I was working on that topology stuff. I believe this (what I just described) is essentially what is going on in a lot of Taylor's ASD work although it isn't quite spelled out in that fashion.

**Comment by jonsterling** [comment]
@freebroccolo oh wow, that sounds really interesting. When you have time, I'd love to learn more about that perspective!

**Comment by ghost** [comment]
Yeah, it's been awhile since I thought about it so I'll have to dig up some stuff and see if I can figure it out again… heh.

**Comment by jonsterling** [comment]
BTW, Ulrik's lecture today might be useful: https://plus.google.com/events/chh8fck2cn8rj49vkdmesuada14
